.github/workflows/observer-e2e.yml

```diff
@@ -0,0 +1,108 @@
+      - 'auto-cadence-upgrade/**'
+      - staging
+      - trying
+      - 'feature/**'
+      - 'v[0-9]+.[0-9]+'
+  pull_request:
+    branches:
+      - master
+      - 'auto-cadence-upgrade/**'
+      - 'feature/**'
+      - 'v[0-9]+.[0-9]+'
+
+jobs:
+  golangci:
+    name: Lint1.45
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.17
+    - name: Checkout repo
+      uses: actions/checkout@v2
+    - name: Download golangci-lint
+      run: bash -c 'curl -L https://github.com/golangci/golangci-lint/releases/download/v1.45.0/golangci-lint-1.45.0-linux-amd64.tar.gz | tar -xzC /opt'
+    - name: Run golangci-lint
+      run: bash -c '(/opt/golangci-lint-1.45.0-linux-amd64/golangci-lint run | sort >/tmp/lint.txt) && cat /tmp/lint.txt && (md5sum </tmp/lint.txt >.github/workflows/lintmd5.txt) && md5sum -c .github/workflows/lintmd5.txt </tmp/lint.txt'
+
+  integration-test:
+    name: Integration Tests
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version:
+          - 1.16
+          - 1.17
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ matrix.go-version}}
+    - name: Checkout repo
+      uses: actions/checkout@v2
+    - name: Build relic
+      run: make crypto_setup_gopath
+    - name: Docker build
+      run: make docker-build-flow
+    - name: Run Observer End To End
+      run: bash -c 'cd integration/localnet/ && make -e OBSERVER=1 init && make start && (docker ps | grep observer) && make stop'
+    - name: Run tests (Bors)
+      if: github.actor == 'bors[bot]'
+      uses: nick-invision/retry@v2
+      with:
+        timeout_minutes: 15
+        max_attempts: 2
+        command: make ci-integration
+  flow-grpc-test:
+    name: Client Tests
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version:
+          - 1.16
+          - 1.17
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ matrix.go-version}}
+    - name: Checkout repo
+      uses: actions/checkout@v2
+    - name: Build relic
+      run: make crypto_setup_gopath
+    - name: Docker build
+      run: make docker-build-flow
+    - name: Install Flow Client
+      run: sh -ci "$(curl -fsSL https://storage.googleapis.com/flow-cli/install.sh)" && mv ~/.local/bin/flow /tmp
+    - name: Set up Localnet
+      run: bash -c 'cd integration/localnet/ && make -e OBSERVER=1 init && make start'
+    - name: Wait 20 seconds to start
+      run: sleep 20
+    - name: Ensure Observer is running unstaked
+      run: docker logs localnet_observer_1_1 2>&1 | grep '"unstaked network"'
+    - name: Get Latest Blocks from Access
+      run: /tmp/flow -f integration/localnet/flow-localnet.json -n access blocks get latest && /tmp/flow -f integration/localnet/flow-localnet.json -n observer blocks get latest || true
```

Comment:
Rather than one-liner cli tests, I think you could use a golang test that takes envvars/args for the access node/observer, and uses the gRPC API to test both are available and in sync.
e.g.
1. Get latest sealed block header from Access Node
2. Get that same block header using its ID from the observer
3. Confirm the header data is the same
4. Confirm the block is not the root block (height > 0)

This would test that:
* the network is running and generating blocks
* the access node and observer are both to syncing the protocol state

Reply:
Thank you for the review. It makes sense.

I will add this together with the first code change, if it is all right.